### PR TITLE
{storage-blob-preview} Support blob service properties with track 2 SDK

### PR DESCRIPTION
**Pull request by Juliehzl**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/storage-blob-preview/azext_storage_blob_preview/_client_factory.py [hunk 1]
```diff
@@ -121,34 +122,3 @@ def cf_blob_sas(cli_ctx, kwargs):
 
     return t_blob_sas(account_name=kwargs.pop('account_name', None),
                       account_key=kwargs.pop('account_key', None))
-
```
Comment (Juliehzl) [inline src/storage-blob-preview/azext_storage_blob_preview/_client_factory.py:124]: remove the code unused

**Comment by yonzhan** [comment]
storage-blob-preview

File: src/storage-blob-preview/azext_storage_blob_preview/_params.py [hunk 2]
```diff
@@ -368,6 +368,26 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
             c.extra('snapshot', snapshot_type)
             c.extra('if_tags_match_condition', tags_condition_type)
 
+    with self.argument_context('storage blob service-properties delete-policy update') as c:
+        c.argument('enable', arg_type=get_enum_type(['true', 'false']), help='Enables/disables soft-delete.')
```
Comment (evelyn-ys) [inline src/storage-blob-preview/azext_storage_blob_preview/_params.py:372]: Why not use `get_three_state_flag()` as `arg_type`?

File: src/storage-blob-preview/azext_storage_blob_preview/operations/blob.py [hunk 3]
```diff
@@ -66,47 +66,35 @@ def set_delete_policy(client, enable=None, days_retained=None):
...
     if error_document_404_path is not None:
-        parameters.static_website.error_document_404_path = error_document_404_path
-    if hasattr(parameters, 'hour_metrics'):
-        kwargs['hour_metrics'] = parameters.hour_metrics
-    if hasattr(parameters, 'logging'):
-        kwargs['logging'] = parameters.logging
-    if hasattr(parameters, 'minute_metrics'):
-        kwargs['minute_metrics'] = parameters.minute_metrics
-    if hasattr(parameters, 'cors'):
-        kwargs['cors'] = parameters.cors
```
Comment (evelyn-ys) [inline src/storage-blob-preview/azext_storage_blob_preview/operations/blob.py:101]: Will `hour_metrics`, `logging`, `minute_metrics` and `cors` not be able to update? Are they not exposed before?

Comment (Juliehzl) [reply src/storage-blob-preview/azext_storage_blob_preview/operations/blob.py:101]: they could but not in here. They could be updated with `logging` related commands. Here we don't plan to expose which is consistent with previous design.

File: src/storage-blob-preview/azext_storage_blob_preview/_params.py (hunk 2 above)
Comment (Juliehzl) [reply src/storage-blob-preview/azext_storage_blob_preview/_params.py:372]: Yes, we could. Here is to align with previous design in track 1 SDK.
https://github.com/Azure/azure-cli/blob/701f1cd0f6f09486193dee977e51a3892cb4bbc0/src/azure-cli/azure/cli/command_modules/storage/_params.py#L647

Comment (Juliehzl) [reply src/storage-blob-preview/azext_storage_blob_preview/_params.py:372]: I just copied all interface definition from main repo and replace the SDK used.

**Review by evelyn-ys — APPROVED** [review]
LGTM

File: src/storage-blob-preview/azext_storage_blob_preview/_params.py (hunk 2 above)
Comment (zhoxing-ms) [inline src/storage-blob-preview/azext_storage_blob_preview/_params.py:372]:
```suggestion
        c.argument('enable', arg_type=get_enum_type(['true', 'false']), help='Enable/disable soft-delete.')
```

File: src/storage-blob-preview/azext_storage_blob_preview/_params.py [hunk 4]
```diff
@@ -368,6 +368,26 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
...
+        c.argument('days_retained', type=int,
+                   help='Number of days that soft-deleted blob will be retained. Must be in range [1,365].')
+
+    with self.argument_context('storage blob service-properties update', min_api='2018-03-28') as c:
+        c.argument('delete_retention', arg_type=get_three_state_flag(), arg_group='Soft Delete',
+                   help='Enables soft-delete.')
```
Comment (zhoxing-ms) [inline src/storage-blob-preview/azext_storage_blob_preview/_params.py:378]:
```suggestion
                   help='Enable soft-delete.')
```

File: src/storage-blob-preview/azext_storage_blob_preview/_params.py [hunk 5]
```diff
@@ -368,6 +368,26 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
...
+        c.argument('delete_retention_period', type=int, arg_group='Soft Delete',
+                   help='Number of days that soft-deleted blob will be retained. Must be in range [1,365].')
+        c.argument('static_website', arg_group='Static Website', arg_type=get_three_state_flag(),
+                   help='Enables static-website.')
```
Comment (zhoxing-ms) [inline src/storage-blob-preview/azext_storage_blob_preview/_params.py:382]:
```suggestion
        c.argument('static_website', arg_group='Static Website', arg_type=get_three_state_flag(),
                   help='Enable static-website.')
```

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**